src/context.rs

```diff
@@ -207,54 +207,63 @@ impl EventCx<'_> {
         if !disable_default {
+            let context_menu = || {
+                let bottom_left = {
+                    let layout = view_state.borrow().layout_rect;
+                    Point::new(layout.x0, layout.y1)
+                };
+
+                let popout_menu = view_state.borrow().popout_menu.clone();
+                show_context_menu(popout_menu?(), Some(bottom_left));
+                Some((EventPropagation::Stop, PointerEventConsumed::Yes))
+            };
+
             match &event {
-                Event::PointerDown(event) => {
+                Event::PointerDown(pointer_event) => {
                     self.app_state.clicking.insert(view_id);
-                    if event.button.is_primary() {
+                    if pointer_event.button.is_primary() {
                         let rect = view_id.get_size().unwrap_or_default().to_rect();
-                        let now_focused = rect.contains(event.pos);
+                        let on_view = rect.contains(pointer_event.pos);
 
-                        if now_focused {
+                        if on_view {
                             if self.app_state.keyboard_navigable.contains(&view_id) {
                                 // if the view can be focused, we update the focus
                                 self.app_state.update_focus(view_id, false);
                             }
-                            if event.count == 2
+                            if pointer_event.count == 2
                                 && view_state
                                     .borrow()
                                     .event_listeners
                                     .contains_key(&EventListener::DoubleClick)
                             {
-                                view_state.borrow_mut().last_pointer_down = Some(event.clone());
+                                view_state.borrow_mut().last_pointer_down =
+                                    Some(pointer_event.clone());
                             }
                             if view_state
                                 .borrow()
                                 .event_listeners
                                 .contains_key(&EventListener::Click)
                             {
-                                view_state.borrow_mut().last_pointer_down = Some(event.clone());
+                                view_state.borrow_mut().last_pointer_down =
+                                    Some(pointer_event.clone());
                             }
 
-                            let bottom_left = {
-                                let layout = view_state.borrow().layout_rect;
-                                Point::new(layout.x0, layout.y1)
+                            #[cfg(target_os = "macos")]
+                            if let Some((ep, pec)) = context_menu() {
+                                return (ep, pec);
                             };
```

Comment:
this should go under secondary click not primary, no?

Reply:
and same thing for not macOS

Reply:
Look at the old diff

Reply:
I guess the naming is wrong/confusing because it's not `context_menu` but `popout_menu` but the function used is actually `show_context_menu`?

Reply:
https://github.com/lapce/floem/blob/67bb86ca17a387e8c00888fc7f79820894061172/src/context.rs#L242-L246

Reply:
ah yeah I was confused by the naming. so it should be on primary pointer down everywhere and probably called popout menu

Reply:
actually not sure about whether pointer down vs up or if there is a different convention on different platforms